addons/website_project/tests/test_website_project.py

```diff
@@ -0,0 +1,55 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.tests.common import HttpCase, tagged
+
+
+@tagged('-at_install', 'post_install')
+class TestWebsiteProject(HttpCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.test_project = cls.env['project.project'].create({'name': 'Project_1'})
+        cls.test_partner = cls.env['res.partner'].create({
+            'name': 'Test Partner',
+            'email': 'test@partner.com',
+        })
+
+    def test_portal_task_submission(self):
+        """ Public user should be able to submit a task"""
+        self.authenticate(None, None)
+        task_data = {
+            'name': "test_task_portal",
+            'email_from': 'test@test.com',
+            'description': 'This test task is created by Portal',
+            'project_id': self.test_project.id,
+            'csrf_token': http.Request.csrf_token(self),
+        }
+        response = self.url_open('/website/form/project.task', data=task_data)
+        task = self.env['project.task'].browse(response.json().get('id'))
+        self.assertTrue(task.exists())
+        self.assertFalse(task.partner_id, "Partner id should be False")
+        self.assertEqual(task.email_cc, 'test@test.com', "email_cc should be same as added on website")
+
+        mail_message = task.message_ids.filtered(lambda m: m.body == '<div class="alert alert-info">/!\\ EXTERNAL SUBMISSION - Customer not verified</div>')
```

Comment:
We could also check the body of the task starts with the "authenticate message". 🙂